addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1308,221 +1308,276 @@ registry.background = SnippetOption.extend({
 
-        var bgImgSize = this.modal.$('> :not(label):not(.o_hidden)').data('value') || 'cover';
-        switch (bgImgSize) {
-            case 'cover':
-                this.$target.css('background-position', this.bgPos.x + ' ' + this.bgPos.y);
-                break;
-            case 'contain':
-                this.$target.addClass('o_bg_img_opt_contain');
-                this.$target.toggleClass('o_bg_img_opt_repeat', this.modal.$('#o_bg_img_opt_contain_repeat').prop('checked'));
-                break;
-            case 'custom':
-                this.$target.addClass('o_bg_img_opt_custom');
-                var sizeX = this.modal.$('#o_bg_img_opt_custom_size_x').val();
-                var sizeY = this.modal.$('#o_bg_img_opt_custom_size_y').val();
-                var posX = this.modal.$('#o_bg_img_opt_custom_pos_x').val();
-                var posY = this.modal.$('#o_bg_img_opt_custom_pos_y').val();
-                this.$target.addClass(this.modal.$('#o_bg_img_opt_custom_repeat').val())
-                            .css({
-                                'background-size': (sizeX ? sizeX + '%' : 'auto') + ' ' + (sizeY ? sizeY + '%' : 'auto'),
-                                'background-position': (posX ? posX + '%' : 'auto') + ' ' + (posY ? posY + '%' : 'auto'),
-                            });
-                break;
+        this._super.apply(this, arguments);
+    },
+    /**
+     * Returns the src value from a css value related to a background image
+     * (e.g. "url('blabla')" => "blabla" / "none" => "").
+     *
+     * @private
+     * @param {string} value
+     * @returns {string}
+     */
+    _getSrcFromCssValue: function (value) {
+        if (value === undefined) {
+            value = this.$target.css('background-image');
+        }
+        var srcValueWrapper = /url\(['"]*|['"]*\)|^none$/g;
+        return value && value.replace(srcValueWrapper, '') || '';
+    },
+    /**
+     * Returns the actual difference between the target's size and the background rendered size.
+     * Background position values in % are actually a percentage of this.
+     *
+     * @private
+     */
+    _getBackgroundDelta: function () {
+        const bgSize = this.$target.css('background-size');
+        if (bgSize !== 'cover') {
+            const [width, height] = bgSize.split(' ').map(v => parseInt(v));
+            return {
+                x: this.$target.outerWidth() - width,
+                y: this.$target.outerHeight() - height,
+            };
+        }
+
+        const renderRatio = Math.max(
+            this.$target.outerWidth() / this.img.naturalWidth,
+            this.$target.outerHeight() / this.img.naturalHeight
+        );
+
+        return {
+            x: this.$target.outerWidth() - Math.round(renderRatio * this.img.naturalWidth),
+            y: this.$target.outerHeight() - Math.round(renderRatio * this.img.naturalHeight),
         }
```

Comment:
It seems my linter was throwing a fatal error on the newly added async function in the color picker because it wasn't configured for the right ES version, quite a few things flew under my radar. I was confused why it was no longer warning me about some stuff, problem solved.